src/close.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{transaction::CloseTransaction, Channel};
+use bitcoin::{Address, Transaction};
+use ecdsa_fun::Signature;
+
+pub struct State0(Channel);
```

Comment:
nitpick: in all other modules we defines the states in between the transition functions.
Here, all the states are defined at the start of the files.

I think it'd be nicer for a reader of our lib to stick to one way.
I have no opinion on which one is better.